### PR TITLE
Add create-react-app boilerplate

### LINKED ISSUE #13: React App Boilerplate
**Is your feature request related to a problem? Please describe.**
We need a boilerplate react app in order to begin large scale development.

**Describe the solution you'd like**
A create-react-app boilerplate, with a basic guide on how to contribute, a scaffolded directory layout, routing, a data fetching method, CSS framework (i.e. tailwind) for our own designs and layouts, a component library for rapid development (NextUI)

**Other alternatives**
NextJS - won't use as it is slightly more complex and hosting may be more complicated.

### PR DESCRIPTION
**Pull request by frasermcc9**
# Description

Adds react to the frontend, as well as several useful tools we will likely want.

Fixes/resolves #13 

- [x] Add react app
- [x] Add basic documentation
- [x] Add directory structure
- [x] Add routing
- [x] Add desired css and ui libraries
- [x] Add a data fetching method
- [x] Add environment variables

## Type of change
Please delete options that are not relevant.

- [x] **New feature** (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:
Leave blank if not applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by j-chad** [comment]
is this supposed to be in draft?

**Comment by j-chad** [comment]
also lets just put the issues into the project and milestone, otherwise everything is duplicated

**Comment by j-chad** [comment]
are we adding unit tests to FE?

**Comment by frasermcc9** [comment]
> are we adding unit tests to FE?

My understanding at the moment was no we won't.

There's not really anything worth testing in this PR anyway

File: packages/frontend/.env.development [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file should be the same for all developers
```
Comment (j-chad) [inline packages/frontend/.env.development:1]: We shouldn't add env files to git. Instead make it a template file and add instructions in README for copying to .env file.
Take a look at #11 for example.

File: packages/frontend/src/components/common/layout/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+interface FooterProps {}
+
+const Footer: React.FC<FooterProps> = () => {
+  return (
+    <div className="opacity-50 w-full mt-8">
+      <div className="w-full text-center p-4">Copyright Team 4</div>
```
Comment (j-chad) [inline packages/frontend/src/components/common/layout/Footer.tsx:8]: Should it be copyright when the project is licensed under MIT?

File: packages/frontend/src/hooks/useApi.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      },
+    });
+    return await parser?.(result);
+  };
+};
+
+/**
+ * Make a request to our API
+ *
+ * @param url the url to the api, i.e. /api/ping
```
Comment (j-chad) [inline packages/frontend/src/hooks/useApi.ts]: just a minor change but should be /api/v1/ping

**Review by j-chad — CHANGES_REQUESTED** [review]

File: packages/frontend/.env.development (hunk 1 above)
Comment (frasermcc9) [reply packages/frontend/.env.development:1]: https://create-react-app.dev/docs/adding-custom-environment-variables/

Sorry I can't direct link to the line but

> .env files should be checked into source control (with the exclusion of .env*.local).

Comment (j-chad) [reply packages/frontend/.env.development:1]: oh ok I guess that makes sense :+1:

**Comment by j-chad** [comment]
help! yarn start:dev does this image

File: packages/frontend/src/components/common/layout/Footer.tsx (hunk 2 above)
Comment (frasermcc9) [reply packages/frontend/src/components/common/layout/Footer.tsx:8]: It was just random text i chucked in 😅 but it led me on an interesting search.

The MIT license is about granting the right to use our copyrighted materials. So we still hold the copyright (which we do implicitly as we created the work)

https://softwareengineering.stackexchange.com/questions/117572/mit-and-copyright

Comment (j-chad) [reply packages/frontend/src/components/common/layout/Footer.tsx:8]: ah very interesting 😄

**Comment by frasermcc9** [comment]
> help! yarn start:dev does this image

whoops

**Comment by j-chad** [comment]
I would also like to see App.tsx moved up to src directory. I think the routes directory is not the best place for it

**Comment by j-chad** [comment]
uh recursive problem is still happening

**Review by j-chad — APPROVED** [review]
LGTM

**Comment by j-chad** [comment]
@frasermcc9 you should add to the contributors table https://github.com/701-T4/platform/wiki/Contributors

**Pull request merged**